modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -141,13 +158,13 @@ MultiSmoothSuperellipsoidIC::computeSuperellipsoidCenters()
       Real ran2 = _random.rand(_tid);
       Real ran3 = _random.rand(_tid);
 
-      newcenter(0) = _bottom_left(0) + ran1*_range(0);
-      newcenter(1) = _bottom_left(1) + ran2*_range(1);
-      newcenter(2) = _bottom_left(2) + ran3*_range(2);
+      newcenter(0) = _bottom_left(0) + ran1 * _range(0);
+      newcenter(1) = _bottom_left(1) + ran2 * _range(1);
+      newcenter(2) = _bottom_left(2) + ran3 * _range(2);
 
-      for (unsigned int j = 0; j < i; j++)
+      for (unsigned int j = start; j < i; j++)
```

Comment:
Well, yeah, but you are still comparing to `_centers[start]` in the first iteration of the loop, which contains an empty point, and then you set `rr` to the cell diagonal so that the test passes always. This whole iteration it just nonsense! Skip it! Start at `start + 1`! And make sure to set `rr` to the right value before the loop.

As I said, this algorithm is not implemented very elegantly. We may need to just rewrite it entirely.